wordpress-mobile/WordPress-iOS
# [News card] Implementation of NewsManager

Implements #9910 

This PR implements the logic that decides when the news card is visible. 

The news card should be visible when:
The card is shown if it has not been dismissed yet
AND
The card is shown on the first Reader filter that users navigate to
AND
If users navigate to another Reader filter, or another screen, the card disappears, but if they navigate back to the filter where it was presented first, it’ll be visible again
AND
If users tap dismiss, the card disappears and will never be displayed again for the same app version

To test:
1. Activate the feature flag
2. To test that old cards will not be visible in new builds of the app, change the version number in `News.strings` to, for example, 10.4
3. Test that the card is not visible anywhere in the reader
4. Change the value of `version` in  `News.strings` back to 10.6
5. Navigate to the reader, to a filter that should not contain a header. The news card should not be visible
6. Navigate to another filter that contains a header. (for example, Discover)
7. The header and the news card should be visible.
8. Navigate away to another filter of Reader. The news card should not be visible, even if there is a header
9. Navigate back to Discover. The news card should be visible as well.
10. Navigate to another tab (for example, Notifications)
11. Go back to Reader > Discover. The news card should still be visible.
12. Dismiss the card. From that moment on the card should not be visible.
